packages/react/src/transactions/infrastructure/ProfileCacheManager.ts

```diff
@@ -0,0 +1,70 @@
+import { FetchPolicy } from '@apollo/client';
+import {
+  GetProfileDocument,
+  GetProfileQuery,
+  GetProfileQueryVariables,
+  LensApolloClient,
+  ProfileFragment,
+  ProfileFragmentDoc,
+  Sources,
+} from '@lens-protocol/api-bindings';
+import { never } from '@lens-protocol/shared-kernel';
+
+import { activeProfileIdentifierVar } from '../../profile/adapters/ActiveProfilePresenter';
+import { FetchProfileArgs, IProfileCacheManager } from '../adapters/IProfileCacheManager';
+
+export class ProfileCacheManager implements IProfileCacheManager {
+  constructor(private readonly client: LensApolloClient, private readonly sources: Sources) {}
+
+  async fetchProfile(args: FetchProfileArgs) {
+    return this.request(args, 'cache-first');
+  }
+
+  async refreshProfile(id: string) {
+    const profile = await this.request({ id }, 'network-only');
+
+    return profile ?? never();
+  }
+
+  private async request(args: FetchProfileArgs, fetchPolicy: FetchPolicy) {
+    const activeProfile = activeProfileIdentifierVar();
+
+    const { data } = await this.client.query<GetProfileQuery, GetProfileQueryVariables>({
+      query: GetProfileDocument,
+      variables: {
+        request: {
+          profileId: args.id,
+          handle: args.handle,
+        },
+        observerId: activeProfile?.id,
+        sources: this.sources,
```

Comment:
do we really need sources to fetch a profile?

Reply:
It's to get these stats counting correctly: https://github.com/lens-protocol/lens-sdk/pull/229#discussion_r1137114156